keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -209,44 +140,40 @@ type CSIVolumeSource struct {
 	// +optional
 	VolumeAttributes map[string]string
 
-	// ControllerPublishSecretRef is a reference to the secret object containing
-	// sensitive information to pass to the CSI driver to complete the CSI
-	// ControllerPublishVolume and ControllerUnpublishVolume calls.
-	// This field is optional, and  may be empty if no secret is required. If the
-	// secret object contains more than one secret, all secret references are passed.
-	// 
-	// +optional
-	ControllerPublishSecretRef *LocalObjectReference
-
-	// NodeStageSecretRef is a reference to the secret object containing sensitive
-	// information to pass to the CSI driver to complete the CSI NodeStageVolume
-	// and NodeStageVolume and NodeUnstageVolume calls.
-	// This field is optional, and  may be empty if no secret is required. If the
-	// secret object contains more than one secret, all secret references are passed.
-	// 
-	// +optional
-	NodeStageSecretRef *LocalObjectReference
-
 	// NodePublishSecretRef is a reference to the secret object containing
 	// sensitive information to pass to the CSI driver to complete the CSI
 	// NodePublishVolume and NodeUnpublishVolume calls.
 	// This field is optional, and  may be empty if no secret is required. If the
 	// secret object contains more than one secret, all secret references are passed.
-	// 
 	// +optional
 	NodePublishSecretRef *LocalObjectReference
 }
 ```
 
-### Secret references
-Secret references declared in an inline CSI volume can only be used with namespaces from pods where they are referenced .  For inline usage, secret references are stored in `LocalObjectReference` values:
-* `LocalObjectReference` do not include a namespace reference.  This is to prevent reference to arbitrary namespace values.
-* The namespace reference will be extracted from the the pod spec at different phases of storage lifecycle by the Kubelet or external CSI compoennts.
-* The Kubelet and external CSI components must ensure secret references can only be used with the namespace from inline pod spec.
+### Driver mode
+To indicate that the driver will support ephemeral inline volume requests, the existing `CSIDriver` object will be extended to include attribute `Mode`.  Currently the only modes that will be supported are `persistent` and `ephemeral`.  
 
+When `CSIDriver.Mode == <not specified>` or when `CSIDriver.Mode == persistent`, the driver will function as normal supporting only PV/PVC-requested volumes and
```

Comment:
Should Mode be defaulted to persistent?

Reply:
`Mode` defaults to <unspecified> so we don't force reconfiguration of existing deployed drivers.  So `unspecified` === `persistent`.